test/suite/functional/Generator/SchemaMapper/AnyOfResponseBodyMapper.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file was generated by docler-labs/api-client-generator.
+ *
+ * Do not edit it manually.
+ */
+
+namespace Test\Schema\Mapper;
+
+use DoclerLabs\ApiClientException\UnexpectedResponseBodyException;
+use Test\Schema\GetExampleResponseBody;
+
+class GetExampleResponseBodyMapper implements SchemaMapperInterface
+{
+    private AnimalMapper $animalMapper;
+
+    private MachineMapper $machineMapper;
+
+    public function __construct(AnimalMapper $animalMapper, MachineMapper $machineMapper)
+    {
+        $this->animalMapper  = $animalMapper;
+        $this->machineMapper = $machineMapper;
+    }
+
+    /**
+     * @throws UnexpectedResponseBodyException
+     */
+    public function toSchema(array $payload): GetExampleResponseBody
+    {
+        $schema = new GetExampleResponseBody();
+
+        try {
+            $schema->setAnimal($this->animalMapper->toSchema($payload));
+        } catch (UnexpectedResponseBodyException $exception) {
+        }
+
+        try {
+            $schema->setMachine($this->machineMapper->toSchema($payload));
+        } catch (UnexpectedResponseBodyException $exception) {
+        }
+        if (empty($schema->toArray())) {
```

Comment:
what if we add a $matches = 0 before all the tries, and inside the try after the $schema->set... we $matches ++

we know when generating this Mapper if it's oneOf or anyOf, so we can already add the code when it's anyOf, we check if ($matches === 0) throw UnexpectedResponseBodyException, when it's oneOf we check if $matches !== 1 throw UnexpectedResponseBodyException

Reply:
and then we can remove this if (empty(toarray()